workspaces/local-cli/src/shared/analytics.ts

```diff
@@ -27,16 +28,24 @@ const analytics = new Analytics(token, {
   flushAt: 1,
 });
 
+const anonIdPromise: Promise<string> = new Promise(async (resolve) => {
```

Comment:
is this redundant?

Reply:
this my pattern for making it run the first time only. 

Later on line 41,47 both await the resolution of this promise. Their parent function can be called many times throughout the lifecycle. It'll wait the first time until it has the token, then all future runs the promise is already resolved so it will go faster.

Reply:
Isn't that the same as saying const x = getOrCreateAnonId(), and await x in both call sites?